Add testing docs for fragment, typePolicies, possibleTypes (#7159)
apollographql/apollo-client: docs/source/development-testing/testing.mdx

For the sake of simplicity, the error case for mutations hasn't been shown here, but testing `useMutation` errors is exactly the same as testing `useQuery` errors: just add an `error` to the mock, fire the mutation, and check the UI for error messages.

## A note on fragment usage, `typePolicies`, and `possibleTypes`

When using fragments with polymorphic types such as the following:
```jsx
// "Dog" supertype can be of type "ShibeInu"
const ShibeFragment = gql`
  fragment ShibeInuFields on Dog {
    ... on ShibeInu {
      tail {
        isCurly
      }
    }
  }
`;

export const GET_DOG_QUERY = gql`
  query GetDog($name: String) {
    dog(name: $name) {
      id
      name
      breed

      ...ShibeInuFields
    }
  }

  ${ShibeFragment}
`;

export const cache = new ApolloClient({
  cache: new InMemoryCache({
    possibleTypes: {
      Dog: ['ShibeInu']
    },
    // suppose you want you key fields for "Dog" to not be simply "id"
    typePolicies: {
      keyFields: {
        Dog: ['name', 'breed']
      }
    }
  })
})
```

> You must configure your cache when testing a component which utilizes fragments in its queries, and pass it to the `<MockedProvider>`.
> 
> `cache` here must be configured with `possibleTypes` and `typePolicies` just the same as it is for your application:

Reviewer: ```suggestion
You must configure your cache when testing a component which utilizes fragments in its queries, and pass it to the `<MockedProvider>` via the `cache={cache}` attribute, configured with `possibleTypes` and `typePolicies` just the same as in your application:
```
